app/src/androidTest/java/org/mozilla/fenix/ui/robots/ReaderViewRobot.kt

```diff
@@ -0,0 +1,263 @@
+                onView(
+                    withId(R.id.mozac_feature_readerview_font_sans_serif)
+                )
+
+            sansSerifButton().click()
+
+            ReaderViewRobot().interact()
+            return Transition()
+        }
+
+        fun toggleSerif(interact: ReaderViewRobot.() -> Unit): Transition {
+            fun serifButton() =
+                onView(
+                    withId(R.id.mozac_feature_readerview_font_serif)
+                )
+
+            serifButton().click()
+
+            ReaderViewRobot().interact()
+            return Transition()
+        }
+
+        fun toggleFontSizeDecrease(interact: ReaderViewRobot.() -> Unit): Transition {
+            fun fontSizeDecrease() =
+                onView(
+                    withId(R.id.mozac_feature_readerview_font_size_decrease)
+                )
+
+            fontSizeDecrease().click()
+
+            ReaderViewRobot().interact()
+            return Transition()
+        }
+
+        fun toggleFontSizeIncrease(interact: ReaderViewRobot.() -> Unit): Transition {
+            fun fontSizeIncrease() =
+                onView(
+                    withId(R.id.mozac_feature_readerview_font_size_increase)
+                )
+
+            fontSizeIncrease().click()
+
+            ReaderViewRobot().interact()
+            return Transition()
+        }
+
+        fun toggleColorSchemeChangeLight(interact: ReaderViewRobot.() -> Unit): Transition {
+            fun toggleLightColorSchemeButton() =
+                onView(
+                    withId(R.id.mozac_feature_readerview_color_light)
+                )
+
+            toggleLightColorSchemeButton().click()
+
+            ReaderViewRobot().interact()
+            return Transition()
+        }
+
+        fun toggleColorSchemeChangeDark(interact: ReaderViewRobot.() -> Unit): Transition {
+            fun toggleDarkColorSchemeButton() =
+                onView(
+                    withId(R.id.mozac_feature_readerview_color_dark)
+                )
+
+            toggleDarkColorSchemeButton().click()
+
+            ReaderViewRobot().interact()
+            return Transition()
+        }
+
+        fun toggleColorSchemeChangeSepia(interact: ReaderViewRobot.() -> Unit): Transition {
+            fun toggleSepiaColorSchemeButton() =
+                onView(
+                    withId(R.id.mozac_feature_readerview_color_sepia)
+                )
+
+            toggleSepiaColorSchemeButton().click()
+
+            ReaderViewRobot().interact()
+            return Transition()
+        }
+    }
+}
+
+fun readerViewRobot(interact: ReaderViewRobot.() -> Unit): ReaderViewRobot.Transition {
+    ReaderViewRobot().interact()
+    return ReaderViewRobot.Transition()
+}
+
+/**
+ * Detects for the blue notification dot in the three dot menu
+ */
+private fun assertReaderViewDetected(visible: Boolean) =
+    onView(
+        nthChildOf(
+            withId(R.id.mozac_browser_toolbar_menu), 2
```

Comment:
The reader view is always the second child of the toolbar menu? Does this work if the toolbar is repositioned to the top as well?

Reply:
The hierarchy of the browser toolbar menu remains the same